tests/e2e/tracer/test_tracer.py

```diff
@@ -46,14 +56,39 @@ def test_lambda_handler_trace_is_visible(basic_handler_fn_arn: str, basic_handle
 
     assert len(trace.get_annotation(key="ColdStart", value=True)) == 1
     assert len(trace.get_metadata(key=handler_metadata_key, namespace=TracerStack.SERVICE_NAME)) == 2
-    assert len(trace.get_metadata(key=handler_metadata_key, namespace=TracerStack.SERVICE_NAME)) == 2
+    assert len(trace.get_metadata(key=method_metadata_key, namespace=TracerStack.SERVICE_NAME)) == 2
     assert len(trace.get_subsegment(name=handler_subsegment)) == 2
     assert len(trace.get_subsegment(name=method_subsegment)) == 2
 
 
+def test_lambda_handler_trace_multiple_functions_same_name(same_function_name_arn: str, same_function_name_fn: str):
+    # GIVEN
+    method_name_1 = "same_function_name.Todos.get_all"
+    method_subsegment_1 = f"## {method_name_1}"
+    method_metadata_key_1 = f"{method_name_1} response"
+
+    method_name_2 = "same_function_name.Comments.get_all"
+    method_subsegment_2 = f"## {method_name_2}"
+    method_metadata_key_2 = f"{method_name_2} response"
+
+    trace_query = data_builder.build_trace_default_query(function_name=same_function_name_fn)
+
+    # WHEN
+    _, execution_time = data_fetcher.get_lambda_response(lambda_arn=same_function_name_arn)
+    data_fetcher.get_lambda_response(lambda_arn=same_function_name_arn)
```

Comment:
no need as we're already asserting on Cold Start in another test.

Reply:
Removed